runtimes/cloudformation/cfnpatcher/patcher.go

```diff
@@ -141,14 +152,21 @@ func applyContainerDefinitionPatch(ctx context.Context, container *gabs.Containe
 		_, err = container.Set([]interface{}{}, "Environment")
 
 		if err != nil {
-		return fmt.Errorf("could not add environment variable container: %w", err)
-	}
+			return fmt.Errorf("could not add environment variable container: %w", err)
+		}
 	}
 
 	for k, v := range patch.EnvironmentVariables {
 		keyValue := make(map[string]interface{})
+
 		keyValue["Name"] = k
-		keyValue["Value"] = v
+
+		if _, ok := cfnInfo.EnvironmentVariables[k]; !ok && configuration.ParameterizeEnvars {
+			keyValue["Value"] = map[string]string{"Ref": getParameterName(k)}
+		} else {
+			keyValue["Value"] = v
+		}
+
```

Comment:
We'll patch the `Parameters` section in the CloudFormation template by adding the parameters related to the envars contained in the recipe. 
Since the template might define a set of envars for the container, we use this check to recognize the envars coming from the recipe (which we need to parameterize).